docs/server/source/production-deployment-template/troubleshoot.rst

```diff
@@ -0,0 +1,77 @@
+Cluster Troubleshooting
+=======================
+
+This page describes some basic issues we have faced while deploying and
+operating the cluster.
+
+1. MongoDB Restarts
+-------------------
+
+We define the following in the ``mongo-ss.yaml`` file:
+
+.. code:: bash
+
+    resources:
+      limits:
+        cpu: 200m
+        memory: 3.5G
+
+So, when the MongoDB cache occupies a memory greater than 3.5GB, it is
+terminated by the ``kubelet``.
+This can usually be verified by logging in to the worker node running MongoDB
+container and looking at the syslog (the ``journalctl`` command should usually
+work). It should show an error message as:
+
+TODO: paste the error message the next time this error happens!
+
+
+2. 502 Bad Gateway Error on Runscope Tests
+------------------------------------------
+
+It means that NGINX could not find the appropriate backed to forward the
+requests to. This typically happens when:
+
+#. MongoDB goes down (as described above) and BigchainDB, after trying for
+   ``BIGCHAINDB_DATABASE_MAXTRIES`` times, gives up. The Kubernetes BigchainDB
+   Deployment then restarts the BigchainDB pod.
+
+#. BigchainDB crashes for some reason. We have seen this happen when updating
+   BigchainDB from one version to the next. This usually means the older
+   connections to the service gets disconnected; retrying the request one more
+   time, forwards the connection to the new instance and succeed.
+
+
+3. Service Unreachable
+----------------------
+
+Communication between Kubernetes Services and Deployments fail in
+v1.6.6 and before due to a trivial key lookup error for non-existent services
+in the ``kubelet``.
+This error can be reproduced by restarting any public facing (that is, services
+using the cloud load balancer) Kubernetes services, and watching the
+``kube-proxy`` failure in its logs.
+The solution to this problem is to restart ``kube-proxy`` on the affected
+worker/agent node. Login to the worker node and run:
+
+.. code:: bash
+
+    docker stop `docker ps | grep k8s_kube-proxy | cut -d" " -f1`
+    
+    docker logs -f `docker ps | grep k8s_kube-proxy | cut -d" " -f1`
+
+This issue is fixed in Kubernetes v1.7.
```

Comment:
Maybe give a link to an issue or pull request about the fix, if you can find one.